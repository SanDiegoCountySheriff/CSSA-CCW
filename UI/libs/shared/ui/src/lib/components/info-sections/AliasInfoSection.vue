<template>
  <v-container class="confirm-info-section rounded mt-5">
    <v-banner class="sub-header font-weight-bold text-xl text-left mb-5">
      {{ $t(' Alias Information: ') }}
      <template #actions>
        <v-btn
          icon
          @click="handleEditRequest"
        >
          <v-icon :color="$vuetify.theme.dark ? 'info' : 'info'">
            mdi-square-edit-outline
          </v-icon>
        </v-btn>
      </template>
    </v-banner>
    <v-row>
      <v-col>
        <AliasTable
          :enable-delete="false"
          :color="props.color"
          :aliases="props.aliasInfo"
        />
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
import AliasTable from '@shared-ui/components/tables/AliasTable.vue'
import { AliasType } from '@shared-utils/types/defaultTypes'
import { useCompleteApplicationStore } from '@shared-ui/stores/completeApplication'
import { useRouter } from 'vue-router/composables'

interface IAliasInfoSectionProps {
  aliasInfo: Array<AliasType>
  color: string
}

const applicationStore = useCompleteApplicationStore()
const router = useRouter()
const props = defineProps<IAliasInfoSectionProps>()

function handleEditRequest() {
  applicationStore.completeApplication.application.currentStep = 1
  router.push({
    path: '/form',
    query: {
      applicationId: applicationStore.completeApplication.id,
      isComplete:
        applicationStore.completeApplication.application.isComplete.toString(),
    },
  })
}
</script>

<style lang="scss">
.confirm-info-section {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
}
</style>
