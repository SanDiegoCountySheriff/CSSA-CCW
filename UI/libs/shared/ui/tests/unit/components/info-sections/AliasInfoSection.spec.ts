/* eslint-disable @typescript-eslint/ban-ts-comment */
import AliasInfoSection from '@shared-ui/components/info-sections/AliasInfoSection.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('AliasInfosection', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = mount(AliasInfoSection, {
      localVue,
      vuetify,
      mocks: {
        $t: text => text,
      },
      propsData: {
        prevLastName: 'last',
        prevFirstName: 'first',
        prevMiddleName: 'middle',
        cityWhereChanged: 'changed',
        stateWhereChanged: 'state',
        courtFileNumber: 'number',
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
