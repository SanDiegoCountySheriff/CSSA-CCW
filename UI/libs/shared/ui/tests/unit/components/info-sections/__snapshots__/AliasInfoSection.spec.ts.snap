// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AliasInfosection Should match the snapshot 1`] = `
<div class="container confirm-info-section rounded container--fluid" prevlastname="last" prevfirstname="first" prevmiddlename="middle" citywherechanged="changed" statewherechanged="state" courtfilenumber="number">
  <div class="v-card v-sheet theme--light">
    <div class="v-card__title ml-3">
      Alias Information
    </div>
    <div class="v-card__text">
      <div class="container container--fluid">
        <div class="v-data-table my-3 theme--light">
          <div class="v-data-table__wrapper">
            <table>
              <thead>
                <tr>
                  <th>Previous Last name</th>
                  <th>Previous First name</th>
                  <th>Previous Middle name</th>
                  <th>City where changed</th>
                  <th>State or region where changed</th>
                  <th>Court file number</th>
                </tr>
              </thead>
              <tbody></tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
